scripts/abi_check.py

```diff
@@ -107,7 +107,7 @@ def _get_clean_worktree_for_git_revision(self, version):
         )
         self.log.debug(worktree_output.decode("utf-8"))
         version.commit = subprocess.check_output(
-            [self.git_command, "rev-parse", worktree_rev],
+            [self.git_command, "rev-parse", "HEAD"],
```

Comment:
Why not set `worktree_rev = "HEAD"` a few lines above?

How should this behave if we've run `worktree_rev = version.revision` since `version.repository` evaluates to false?

Reply:
If we set `worktree_rev = "HEAD"` earlier, then checking out the worktree would just check out the existing branch, rather than the specified one.

It'll behave in the same manner as at this point we've already checked out the worktree. The logic above is just for handling a branch from a remote or one known locally.

Reply:
Should we do this for the other FETCH_HEAD line as well?

Reply:
No, this is the only place that does the rev-parse operation.